pkg/agent/manager/cache/cache_test.go

```diff
@@ -75,6 +78,45 @@ func TestMatchingIdentities(t *testing.T) {
 	}, identities)
 }
 
+func TestRegistrationEntryMetrics(t *testing.T) {
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+	metrics := mock_telemetry.NewMockMetrics(mockCtl)
+
+	log, _ := test.NewNullLogger()
+	cache := New(log, "spiffe://domain.test", bundleV1, metrics)
+
+	// populate the cache with FOO and BAR without SVIDS
+	foo := makeRegistrationEntry("FOO", "A")
+	bar := makeRegistrationEntry("BAR", "B")
+	update := &CacheUpdate{
+		Bundles:             makeBundles(bundleV1),
+		RegistrationEntries: makeRegistrationEntries(foo, bar),
+	}
+
+	// Add two new entries
+	metrics.EXPECT().IncrCounterWithLabels([]string{telemetry.CacheManager, telemetry.RegistrationEntry, telemetry.Create}, gomock.Any(),
```

Comment:
I'm not a huge fan of asserting the metrics were produced by ensure the `metrics.Metrics` interface was called with certain arguments. This strongly couples the test to the internals of `IncrRegistrationEntryCreatedCounter` and family. I'd much rather see something like this:

```
func TestStuff(t *testing.T) {
    ...
    expected := fakemetrics.New()
    IncrRegistrationEntryCreatedCounter(expected, foo.SpiffeId)

    actual := fakemetrics.New()
    cache := New(..., actual)
    ....
    cache.Update(update, nil)

    require.Equal(t, expected.AllMetrics(), actual.AllMetrics())
}
```

Where fakemetrics is a simple in-memory `metrics.Metrics` implementation that just records what it is called with.